scripts/zkai_native_attention_mlp_lifting_ablation_gate.py

```
@@ -0,0 +1,747 @@
+        raise LiftingAblationError(f"{label} is not JSON: {err}") from err
+
+
+def read_json(path: pathlib.Path, max_bytes: int, label: str) -> Any:
+    payload, _size, _raw = read_json_with_size(path, max_bytes, label)
+    return payload
+
+
+def sha256_file(path: pathlib.Path, max_bytes: int, label: str) -> tuple[str, int]:
+    _payload, size, raw = read_json_with_size(path, max_bytes, label)
+    if len(raw) != size:
+        raise LiftingAblationError(f"{label} changed while hashing")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)